packages/compartment-mapper/src/compartment-map-transforms/transforms.js

```diff
@@ -0,0 +1,351 @@
+/* eslint-disable no-continue */
+
+/**
+ * Compartment Map Transform implementations
+ *
+ * @module
+ */
+
+import {
+  ATTENUATORS_COMPARTMENT,
+  ENTRY_COMPARTMENT,
+  WILDCARD_POLICY_VALUE,
+} from '../policy-format.js';
+
+/**
+ * @import {CompartmentDescriptor,
+ *   CompartmentMapTransformFn,
+ *   CompartmentModuleDescriptorConfiguration,
+ *   FileUrlString,
+ *   LogFn,
+ *   ModuleDescriptorConfiguration,
+ *   PackageCompartmentDescriptor,
+ *   PackageCompartmentDescriptorName,
+ *   PackageCompartmentMapDescriptor,
+ *   PropertyPolicy,
+ *   SomePackagePolicy,
+ * } from '../types.js'
+ */
+
+/**
+ * Dummy logger
+ * @type {LogFn}
+ */
+const noop = () => {};
+
+const { quote: q } = assert;
+const { entries, values, freeze } = Object;
+const { isArray } = Array;
+
+/**
+ * Type guard for a {@link PropertyPolicy}
+ *
+ * @param {unknown} value
+ * @returns {value is PropertyPolicy}
+ */
+const isPropertyPolicy = value =>
+  !!value &&
+  typeof value === 'object' &&
+  !isArray(value) &&
+  values(value).every(item => typeof item === 'boolean');
+
```

Comment:
(nit) Split each transform into a separate file, aggregate builtin transforms in index.js